iroha_p2p/tests/mod.rs

```diff
@@ -0,0 +1,65 @@
+#![allow(clippy::unwrap_used)]
+use iroha_actor::{broker::*, prelude::*};
+use iroha_logger::{config::LoggerConfiguration, info};
+use parity_scale_codec::{Decode, Encode};
+use tokio::time::Duration;
+use ursa::{encryption::symm::prelude::ChaCha20Poly1305, kex::x25519::X25519Sha256};
+
+use iroha_p2p::*;
+use iroha_p2p::peer::PeerId;
+
+#[derive(iroha_actor::Message, Clone, Debug, Decode, Encode)]
+struct TestMessage(String);
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+async fn network_create() {
+    drop(iroha_logger::init(LoggerConfiguration::default()).unwrap());
+    info!("Starting network tests...");
+    let broker = Broker::new();
+    let net = Network::<TestMessage, X25519Sha256, ChaCha20Poly1305>::new(
+        broker.clone(),
+        String::from("127.0.0.1:4433"),
+    )
+    .await
+    .unwrap();
+    drop(net.start().await);
+    tokio::time::sleep(Duration::from_millis(100)).await;
+
+    info!("Connecting to peer...");
+    let peer1 = PeerId {
+        address: String::from("127.0.0.1:4433"),
+        public_key: None,
+    };
+    broker.issue_send(Connect { id: peer1.clone() }).await;
```

Comment:
I mean, can you explain a flow in a bit more detail:
1. You create a network that listens on :4433
2. You ask a network to connect a peer at :4433, but this is the address of the peer that started this network already, so why do you connect to it?

Reply:
Ah, yep. For this test we create a peer that is connecting to ourselves. It really doesn't matter where to connect.
The main concern is how to check that listening peer and connecting peer have same sent/received message count, for example.